src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Locale;
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_VALUE_OF_PARAM = 25;
+    public static final int NUMBER_OF_FIGURES = 5;
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(NUMBER_OF_FIGURES);
+        switch (index) {
+            case 0:
+                return new Circle(new ColorSupplier().getRandomColor(),
+                        new Random().nextInt(MAX_VALUE_OF_PARAM));
```

Comment:
let's create `new ColorSupplier()` and `new Random()` only once - use local variables